description: adjust search paths for addon installers, but don't download installers
author: Michael Gilbert <mgilbert@debian.org>
author: Jens Reyer <jre.winesim@gmail.com>

--- a/dlls/appwiz.cpl/addons.c
+++ b/dlls/appwiz.cpl/addons.c
@@ -209,8 +209,6 @@ static enum install_res install_from_uni
         return INSTALL_FAILED;
 
     memcpy(file_path, dir, len);
-    if(len && file_path[len-1] != '/' && file_path[len-1] != '\\')
-        file_path[len++] = '/';
     if(*subdir) {
         strcpy(file_path+len, subdir);
         len += strlen(subdir);
@@ -320,11 +318,11 @@ static enum install_res install_from_def
         return INSTALL_NEXT;
     }
 
-    ret = install_from_unix_file(package_dir, addon->subdir_name, addon->file_name);
+    ret = install_from_unix_file("/usr/share/wine-", addon->subdir_name, addon->file_name);
     heap_free(dir_buf);
 
     if (ret == INSTALL_NEXT)
-        ret = install_from_unix_file(INSTALL_DATADIR "/wine/", addon->subdir_name, addon->file_name);
+        ret = install_from_unix_file(INSTALL_DATADIR "/", addon->subdir_name, addon->file_name);
     if (ret == INSTALL_NEXT && strcmp(INSTALL_DATADIR, "/usr/share"))
         ret = install_from_unix_file("/usr/share/wine/", addon->subdir_name, addon->file_name);
     return ret;
@@ -795,9 +793,8 @@ BOOL install_addon(addon_t addon_type)
      */
     if (install_from_registered_dir() == INSTALL_NEXT
         && install_from_default_dir() == INSTALL_NEXT
-        && install_from_cache() == INSTALL_NEXT
-        && (url = get_url()))
-        DialogBoxW(hInst, addon->dialog_template, 0, installer_proc);
+        && install_from_cache() == INSTALL_NEXT)
+        {};
 
     heap_free(url);
     url = NULL;
